en/docs/develop/api-policies/create-a-policy.md

```diff
@@ -0,0 +1,291 @@
+        - **error** - Return an error if you want to terminate the mediation flow and transfer control to the fault flow. The fault flow would then construct an error response and send it to the client.
+        - **()** - Return () to signal the successful completion of the policy. On completion of the policy, the proxy executes the next policy in the sequence.
+
+## Step 3: Publish the policy
+
+You can publish mediation policies to Ballerina Central. When you deploy the API after attaching policies, Choreo pulls these packages from Ballerina Central and bundles them into the interceptor application under the hood. You can find more details on building and deploying an interceptor application [here](https://wso2.com/choreo/docs/develop/components/api-proxies/policies/). To use policies in your APIs, follow the steps below to publish them as public packages:
+
+1. Package the policy before publishing it to Ballerina Central by issuing the following command;
+    ``` 
+        bal pack 
+    ```
+2. Publish the package to Ballerina Central by issuing the following command;
+
+    ``` 
+        bal push 
+    ```
+
+Once you publish the package, it will be listed in the policy list in the Choreo Console as below:
+
+![Added policy](../../assets/img/develop/api-policy/added-policy.png){.cInlineImage-small}
+
+## Write unit tests
+
+Writing unit tests to test policy functions is similar to writing unit tests for a regular [Ballerina function](https://ballerina.io/learn/test-ballerina-code/test-a-simple-function/). 
+
+Let’s write a unit test for the validateRequestHeader function.
+
+```
+import ballerina/http;
+import choreo/mediation;
+import ballerina/test;
+
+
+@test:Config {}
+public function testRequestHeaderValidationFailure() {
+  http:Request req = new;
+  http:Response|false|error|() result = validateRequestHeader(createContext("get", "/test"), req, "testHeader", "test");
+
+
+  if !(result is http:Response) {
+    test:assertFail("Expected http:Response, found " + (typeof result).toString());
+  }
+
+
+  test:assertEquals(result.statusCode, http:STATUS_BAD_REQUEST, "Status code mismatch");
+}
+
+
+function createContext(string httpMethod, string resPath) returns mediation:Context {
+   mediation:ResourcePath originalPath = checkpanic mediation:createImmutableResourcePath(resPath);
+   mediation:Context originalCtx =
+               mediation:createImmutableMediationContext(httpMethod, originalPath.pathSegments(), {}, {});
+   mediation:ResourcePath mutableResPath = checkpanic mediation:createMutableResourcePath(resPath);
+   return mediation:createMutableMediationContext(originalCtx, mutableResPath.pathSegments(), {}, {});
+}
+
+```
+
+As the policy function is modifying the same request/response/context instance that you pass to it, to verify the changes, you have to verify the request/response/context instance after calling the policy function.
```

Comment:
Suggestion:
"The policy function modifies the same request/response/context instance that you pass to it. Check the request/response/context instance after calling the policy function to verify these changes."

Reply:
done